python/pyarrow/dataset.py

```diff
@@ -153,15 +153,14 @@ def partitioning(schema=None, field_names=None, flavor=None):
 
 def _ensure_fs(filesystem, path):
     # Validate or infer the filesystem from the path
-    from pyarrow.fs import FileSystem, LocalFileSystem
+    from pyarrow.fs import FileSystem, LocalFileSystem, FileType
 
     if filesystem is None:
-        try:
+        # first check if the file exists as a local (relative) file path
+        filesystem = LocalFileSystem()
+        infos = filesystem.get_target_infos([path])[0]
```

Comment:
Hmm, I don't understand why the Python filesystem wrapper doesn't get a `get_target_info` taking a single path argument. The `[0]` dance is certainly not pretty :-)